torch/_subclasses/schema_check_mode.py

```diff
@@ -124,6 +124,11 @@ def parse_metadata(e):
                             raise RuntimeError(f'Argument {name} is not defined to alias output but was aliasing')
                         else:
                             self.aliasing.append(Aliasing(func._schema.name, name, f"output_{j}"))
+                    for i, inp in enumerate(tree_flatten(after)[0]):
+                        if inp is tuple_out[j]:
+                            raise RuntimeError(f"""\
+Dispatcher operators below autograd are not allowed to directly return inputs.
+However, we found that `outputs[{str(j)}] is {name}""")
```

Comment:
cc @albanD (making sure this check sounds reasonable to you)

Reply:
Sounds fair to me!